### PR TITLE
ZNTA-779: configuration: locales mappings

### PR DESCRIPTION
**Pull request by sundeep-co-in**
* Code refactor, introducing `ProjectContext`
* Implement `LocaleResources` REST APIs

*A round of complete test is requested for this change.*

@definite Please have a look.

### CODE REVIEW

File: test/test_parseconfig.py [hunk 1]
```diff
@@ -46,12 +46,12 @@ def test_user_config(self):
...
     def test_project_config(self):
         project_config = self.config.read_project_config("./testfiles/zanata.xml")
-        self.assertEqual(project_config['project_url'], "http://localhost:8080/zanata/")
+        self.assertEqual(project_config['url'], "http://localhost:8080/zanata/")
         self.assertEqual(project_config['project_id'], "test-project")
         self.assertEqual(project_config['project_version'], "1.0")
-        self.assertEqual(project_config['locale_map'], {"zh-CN": "zh-Hans"})
-        self.assertEqual(project_config['src_dir'], "/home/user/project/source")
-        self.assertEqual(project_config['trans_dir'], "/home/user/project/target")
+        # self.assertEqual(project_config['locale_map'], {"zh-CN": "zh-Hans"})
```
Comment (definite) [inline test/test_parseconfig.py]: It looks like you were trying to disable test on locale_map, why is that?

Comment (sundeep-co-in) [inline test/test_parseconfig.py]: Actually, need to comment `locale_map` here https://github.com/sundeep-co-in/zanata-python-client/blob/STAGE/test/testfiles/zanata.xml#L7 so, that context can go for `remote locale_map` https://github.com/sundeep-co-in/zanata-python-client/blob/STAGE/zanataclient/context.py#L247

As, both `test_parseconfig` and `test_context` https://github.com/sundeep-co-in/zanata-python-client/blob/STAGE/test/test_context.py#L40 pointing to same test file `testfiles/zanata.xml`, need to disable that.

**Comment by definite** [comment]
:+1:

**Comment by definite** [comment]
:+1:

**Pull request merged**